lib/sprig/reap/record.rb

```diff
@@ -11,14 +11,14 @@ def initialize(record, model)
     end
 
     def attributes
-      @attributes ||= model.attributes.delete_if { |a| a == "id" }
+      @attributes ||= model.attributes.delete_if { |a| a == "id" } | model.associations.map(&:has_and_belongs_to_many_attr).compact
```

Comment:
The only difference is that `|` eliminates duplicates for you, so really depends on your use case.